src/writer/xapianIndexer.cpp

```diff
@@ -65,12 +65,27 @@ XapianIndexer::~XapianIndexer()
   }
 }
 
+/*
+ * `valuesmap` is a metadata associated with the Xapian database. We are using it
+ * to attach slot numbers of each document in the index to the value they are storing.
+ * These values and slot numbers are used in collapsing, filtering etc.
+ *
+ * Title index:
+ * Slot 0: Title of the article. Used in collapsing articles with same name.
+ * Slot 1: path/redirectPath of the article. Used in collapsing duplicates(redirects).
+ *
+ * Fulltext Index:
+ * Slot 0: Title of the article. Used in collapsing articles with same name.
+ * Slot 1: Word count of the article.
+ * Slot 2: Geo position of the article. Used for geo-filtering.
+ */
+
 void XapianIndexer::indexingPrelude()
 {
   writableDatabase = Xapian::WritableDatabase(indexPath + ".tmp", Xapian::DB_CREATE_OR_OVERWRITE);
   switch (indexingMode) {
     case IndexingMode::TITLE:
-      writableDatabase.set_metadata("valuesmap", "title:0");
+      writableDatabase.set_metadata("valuesmap", "title:0;path:1");
```

Comment:
We are not storing the path of the entry. So "path" is maybe not the good name after all.
We may name it "targetPath"/"finalPath" (if we want to describe what is in the value) or "collapseKey"(if we want to describe how we use the value).

I would go with "targetPath". (It would be "finalPath" if we were resolving redirect chain).

Reply:
Yeah that makes sense. I should change `path` to `targetPath`.